addons/refuel/functions/fnc_takeNozzle.sqf

```diff
@@ -55,10 +55,13 @@ if (isNull _nozzle) then { // func is called on fuel truck
             _newNozzle attachTo [_unit, [-0.02,0.05,-0.12], "righthandmiddle1"];
             _unit setVariable [QGVAR(nozzle), _newNozzle, true];
 
-            private _rope = ropeCreate [_target, _endPosOffset, _newNozzle, [0, -0.20, 0.12], REFUEL_HOSE_LENGTH];
+            if (_target isKindOf "AllVehicles") then {
+                // Currently ropeCreate requires its first parameter to be a real vehicle
+                private _rope = ropeCreate [_target, _endPosOffset, _newNozzle, [0, -0.20, 0.12], REFUEL_HOSE_LENGTH];
+                _newNozzle setVariable [QGVAR(rope), _rope, true];
+            };
```

Comment:
We can either support those objects (i.e. gas pump) and the player has to "live without the rope", or we exclude them from refuel.

Since I am in hope that BIS will fix this problem, I'd like to include those objects and find ways to make refuel work.
Especially if it creates problems like mentioned in #3310, where the CSAT slingload fuel tank is working but the NATO isn't.

I'm also under the impression that calling `createRope` on a non vehicle seems to exit/break the current scope as `QGVAR(source)` was not set on the nozzle when interacting with a `B_Slingload_01_Fuel_F` object. I received the nozzle, but could not return it.